### PR TITLE
Fix#151 Last Sync Info displayed

### PR DESCRIPTION
**Pull request by pawanharariya**
All previous suggestions are incorporated.
1. dateTime is stored as time stamp
2. All constants are declared in seperate file
3. PropertiesSingleton is used instead of SharedPreference
4. Whitespace changes removed

### CODE REVIEW

**Comment by wbrunette** [comment]
runtests

**Comment by wbrunette** [comment]
replaces #183

**Comment by wbrunette** [comment]
replaces: #181

**Comment by pawanharariya** [comment]
fixes issue#201

**Comment by wbrunette** [comment]
runtests

**Review by wbrunette — APPROVED** [review]

File: services_app/src/main/java/org/opendatakit/services/utilities/Constants.java [hunk 1]
```diff
@@ -11,4 +11,5 @@
     public static final String ODK_CLIENT_API_VERSION = "2";
     public static final String KEY_LAST_SYNC_INFO = "lastSyncInfo";
     public static final String DATE_TIME_FORMAT = "dd-MM-yyyy HH:mm:ss";
+    public static final String KEY_SYNC_ATTACHMENT_STATE = "syncTypeSpinnerPreference";
```
Comment (linl33) [inline services_app/src/main/java/org/opendatakit/services/utilities/Constants.java:14]: This and `KEY_LAST_SYNC_INFO` should be in https://github.com/odk-x/androidlibrary/blob/development/androidlibrary_lib/src/main/java/org/opendatakit/properties/CommonToolProperties.java.

Please open a pull request to https://github.com/odk-x/androidlibrary first.

File: services_app/src/main/java/org/opendatakit/services/sync/actions/fragments/SyncFragment.java [hunk 2]
```diff
@@ -440,7 +450,22 @@ void syncCompletedAction(IOdkSyncServiceInterface syncServiceInterface) throws R
     perhapsEnableButtons();
     updateInterface();
   }
-
+  void saveLastSyncInfo(SyncStatus status){
```
Comment (linl33) [inline services_app/src/main/java/org/opendatakit/services/sync/actions/fragments/SyncFragment.java:452]: This method could be `private`

**Review by linl33 — CHANGES_REQUESTED** [review]
@pawanharariya Thanks for the changes. Please see the comments.

Could you send a separate pull request for sync attachment spinner changes? In general, each pull request should address 1 issue.

Fixes odk-x/tool-suite-x#155

File: services_app/src/main/java/org/opendatakit/services/utilities/Constants.java (hunk 1 above)
Comment (pawanharariya) [reply services_app/src/main/java/org/opendatakit/services/utilities/Constants.java:14]: Thanks for pointing these.
Should I name the value to "common.lastSyncInfo" or "services.lastSyncInfo" and same of spinner. How is the naming being done?

**Comment by wbrunette** [comment]
@pawanharariya please open your androidlibrary PR against the "development" branch NOT the "master" branch. I caught it right after I merged so I reverted the original PR merge.

**Comment by pawanharariya** [comment]
@linl33  sorry for inconvenience. All required changes are done:
1. Unused import util.log removed
2. Methods displayLastSyncInfo and saveLastSyncInfo declared private
3. KEYS are moved to androidlibrary

**Comment by linl33** [comment]
runtests

**Review by linl33 — APPROVED** [review]

**Comment by linl33** [comment]
@pawanharariya Please resolve the merge conflict then we can merge this pull request.

**Comment by pawanharariya** [comment]
@linl33 merge conflicts have been resolved. Sorry for inconvenience.

**Comment by wbrunette** [comment]
Fixes https://github.com/odk-x/tool-suite-X/issues/155

**Pull request merged**